src/createRollupPrepareTransactionRequest.ts

```diff
@@ -72,6 +73,25 @@ export async function createRollupPrepareTransactionRequest<TChain extends Chain
     }
   }
 
+  const arbOSVersion = chainConfig.arbitrum.InitialArbOSVersion;
+  const wasmModuleRoot = params.config.wasmModuleRoot;
+
+  if (arbOSVersion === 30 || arbOSVersion === 31) {
+    throw new Error(
+      `ArbOS ${arbOSVersion} is not supported. Please set the ArbOS version to 32 or later by updating "arbitrum.InitialArbOSVersion" in your chain config.`,
+    );
+  }
+
+  if (isKnownWasmModuleRoot(wasmModuleRoot)) {
+    const consensusRelease = getConsensusReleaseByWasmModuleRoot(wasmModuleRoot);
+
+    if (arbOSVersion > consensusRelease.maxArbOSVersion) {
+      throw new Error(
+        `Consensus v${consensusRelease.version} does not support ArbOS ${arbOSVersion}. Please update your "wasmModuleRoot" to that of a Consensus version compatible with ArbOS ${arbOSVersion}.`,
+      );
+    }
+  }
```

Comment:
If it's a known wasm module root, make sure it's compatible with the selected ArbOS version